shell/platform/windows/flutter_windows.cc

```diff
@@ -97,84 +97,60 @@ static FLUTTER_API_SYMBOL(FlutterEngine)
   return engine;
 }
 
-bool FlutterDesktopInit() {
-  return true;
-}
-
-void FlutterDesktopTerminate() {}
-
-FlutterDesktopWindowControllerRef FlutterDesktopCreateWindow(
+FlutterDesktopViewControllerRef FlutterDesktopCreateView(
     int initial_width,
     int initial_height,
-    const char* title,
     const char* assets_path,
     const char* icu_data_path,
     const char** arguments,
     size_t argument_count) {
-  FlutterDesktopWindowControllerRef state =
-      flutter::Win32FlutterWindow::CreateWin32FlutterWindow(
-          title, 10, 10, initial_width, initial_height);
+  FlutterDesktopViewControllerRef state =
+      flutter::Win32FlutterWindow::CreateWin32FlutterWindow(initial_width,
+                                                            initial_height);
 
-  auto engine = RunFlutterEngine(state->window.get(), assets_path,
-                                 icu_data_path, arguments, argument_count);
+  auto engine = RunFlutterEngine(state->view.get(), assets_path, icu_data_path,
+                                 arguments, argument_count);
 
   if (engine == nullptr) {
     return nullptr;
   }
 
-  state->window->SetState(engine);
-
-  // Trigger an initial size callback to send size information to Flutter.
-  state->window->SendWindowMetrics();
+  state->view->SetState(engine);
+  state->engine = engine;
 
   return state;
 }
 
-void FlutterDesktopDestroyWindow(FlutterDesktopWindowControllerRef controller) {
-  FlutterEngineShutdown(controller->engine);
-  delete controller;
+void FlutterDesktopProcessMessages() {
+  __FlutterEngineFlushPendingTasksNow();
 }
 
-void FlutterDesktopWindowSetHoverEnabled(FlutterDesktopWindowRef flutter_window,
-                                         bool enabled) {
-  // todo either implement or remove once embedder project has moved
+long FlutterDesktopGetHWNDFromView(FlutterDesktopViewRef flutter_window) {
+  return (long)((FlutterDesktopView*)flutter_window)->window->GetWindowHandle();
 }
 
-void FlutterDesktopWindowSetTitle(FlutterDesktopWindowRef flutter_window,
-                                  const char* title) {
-  // todo either implement or remove
-}
-
-void FlutterDesktopWindowSetIcon(FlutterDesktopWindowRef flutter_window,
-                                 uint8_t* pixel_data,
-                                 int width,
-                                 int height) {
-  // todo either implement or remove
-}
-
-void FlutterDesktopRunWindowLoop(FlutterDesktopWindowControllerRef controller) {
-  controller->window->FlutterMessageLoop();
-
-  FlutterDesktopDestroyWindow(controller);
+void FlutterDesktopDestroyView(FlutterDesktopViewControllerRef controller) {
```

Comment:
These two functions say they operate on views, but one is using a view and one is using a view controller, which seems confusing.

Stepping back slightly: is there still a reason to have a "view" object separate from the view controller like there was with window/window_controller? The main reason for the separation in the GLFW API is that plugins needed to be able to operate on a window, but shouldn't have access to the entire controller, so I split them. I would expect the HWND to serve that purpose here, so would it make sense to just have the view controller in the API, and no view? (Internally it may still be useful to have a separation, but it's not clear that needs to be reflected to the C API).

Reply:
This does make sense.  Next commit will combine them.